wordle-web/src/scripts/wordleGame.ts

```diff
@@ -57,4 +57,40 @@ export class WordleGame {
       // The game is over
     }
   }
+
+  getAvailableWords() {
+    const availableWords: string[] = []
+    const secretWordChars = this.secretWord.split('')
+    const wordList = WordsService.getWordsList()
+
+    for (const word of wordList) {
+      const chars = word.split('')
+      let isAvailable = true
+
+      for (let i = 0; i < chars.length; i++) {
+        const char = chars[i]
+        const index = secretWordChars.indexOf(char)
+
+        if (index === -1) {
+          isAvailable = false
+          break
+        } else {
+          secretWordChars.splice(index, 1)
+        }
+      }
+
+      if (isAvailable) {
+        availableWords.push(word)
+      }
+
+      if (secretWordChars.length === 0) {
+        break
+      } else {
+        secretWordChars.splice(0)
+        secretWordChars.push(...this.secretWord.split(''))
+      }
+    }
+
+    return availableWords
+  }
```

Comment:
A good start to refactor this to work correctly would be to check against guesses rather than the secret word. What helps me is to comment (almost excessively) every major line as I go, to read in plain english what steps are doing what. Also, more descriptive naming than `char` and `chars` might make things more clear for you while you're writing it. Just advice, not trying to criticize.

Reply:
Great suggestion! Thanks!